Sources/libreprl/libreprl-windows.c

```diff
@@ -0,0 +1,439 @@
+    if (script_length > REPRL_MAX_DATA_SIZE)
+        return reprl_error(ctx, "Script too large");
+
+    if (fresh_instance && ctx->hChild != INVALID_HANDLE_VALUE)
+        reprl_terminate_child(ctx);
+
+    SetFilePointer(ctx->data_out->hFile, 0, 0, FILE_BEGIN);
+    SetFilePointer(ctx->data_in->hFile, 0, 0, FILE_BEGIN);
+    if (ctx->child_stdout)
+        SetFilePointer(ctx->child_stdout->hFile, 0, 0, FILE_BEGIN);
+    if (ctx->child_stderr)
+        SetFilePointer(ctx->child_stderr->hFile, 0, 0, FILE_BEGIN);
+
+    if (ctx->hChild == INVALID_HANDLE_VALUE) {
+        int r = reprl_spawn_child(ctx);
+        if (r)
+            return r;
+    }
+
+    // Copy the script to the data channel.
+    memcpy(ctx->data_out->mapping, script, script_length);
+
+    // Tell child to execute the script.
+    DWORD dwBytesWritten;
+    if (!WriteFile(ctx->hControlWrite, "exec", 4, &dwBytesWritten, NULL) || dwBytesWritten != 4) {
+        DWORD dwExitCode;
+        if (GetExitCodeProcess(ctx->hChild, &dwExitCode) != STILL_ACTIVE) {
+            reprl_child_terminated(ctx);
+            return reprl_error(ctx, "Child unexpected terminated with status %u between executions", dwExitCode);
+        }
+        return reprl_error(ctx, "Failed to send command to child process: %d", GetLastError());
+    }
+    if (!WriteFile(ctx->hControlWrite, &script_length, sizeof(script_length), &dwBytesWritten, NULL) || dwBytesWritten != sizeof(script_length)) {
+        DWORD dwExitCode;
+        if (GetExitCodeProcess(ctx->hChild, &dwExitCode) != STILL_ACTIVE) {
+            reprl_child_terminated(ctx);
+            return reprl_error(ctx, "Child unexpected terminated with status %u between executions", dwExitCode);
+        }
+        return reprl_error(ctx, "Failed to send command to child process: %d", GetLastError());
+    }
+
+    switch (WaitForSingleObject(ctx->hChild, timeout / 1000)) {
+    case WAIT_TIMEOUT:
+        reprl_terminate_child(ctx);
+        return 1 << 16;
+    case WAIT_OBJECT_0:
+        break;
+    default:
+        return reprl_error(ctx, "WaitForSingleObject error: %d", GetLastError());
+    }
+
+    int status;
+    DWORD dwBytesRead;
+    if (!ReadFile(ctx->hControlRead, &status, sizeof(status), &dwBytesRead, NULL))
+        return reprl_error(ctx, "unable to read from control pipe: %d", GetLastError());
+    if (dwBytesRead == sizeof(status))
+        return status & 0xffff;
+    DWORD dwExitCode;
+    BOOL bSuccess;
+    bSuccess = GetExitCodeProcess(ctx->hChild, &dwExitCode);
+
+    reprl_terminate_child(ctx);
+    if (bSuccess)
+        return status & 0xffff;
+    return reprl_error(ctx, "child in weird state after execution");
+}
+
+/// The 32bit REPRL exit status as returned by reprl_execute has the following format:
+///     [ 00000000 | did_timeout | exit_code | terminating_signal ]
+/// Only one of did_timeout, exit_code, or terminating_signal may be set at one time.
+int RIFSIGNALED(int status)
```

Comment:
Would it make sense to move these 5 functions into a libreprl-common.c file?

Reply:
I was actually thinking that these functions don't make sense to have at all - we should put them into the header as `static inline`.  I didn't want to merge that into this change, but intended to do that as a follow up.

Reply:
#281 does that :)